josh-cli/src/bin/josh.rs

```diff
@@ -803,6 +854,350 @@ fn handle_push(args: &PushArgs) -> anyhow::Result<()> {
+        .args(&["fetch", &args.url, &target])
+        .output()
+        .context("Failed to execute git fetch")?;
+
+    if !output.status.success() {
+        return Err(anyhow::anyhow!(
+            "git fetch failed: {}",
+            String::from_utf8_lossy(&output.stderr)
+        ));
+    }
+
+    // Get the commit SHA from FETCH_HEAD
+    let fetch_head = repo
+        .find_reference("FETCH_HEAD")
+        .context("Failed to find FETCH_HEAD")?;
+    let fetch_commit = fetch_head
+        .peel_to_commit()
+        .context("Failed to get FETCH_HEAD commit")?;
+    let actual_commit_sha = fetch_commit.id();
+
+    // Get the current HEAD commit
+    let head_ref = repo.head().context("Failed to get HEAD")?;
+    let head_commit = head_ref
+        .peel_to_commit()
+        .context("Failed to get HEAD commit")?;
+    let head_tree = head_commit.tree().context("Failed to get HEAD tree")?;
+
+    // Create the JoshLinkFile with the actual commit SHA
+    let link_file = JoshLinkFile {
+        remote: args.url.clone(),
+        branch: target.to_string(),
+        filter: filter_obj,
+        commit: Oid::from(actual_commit_sha),
+    };
+
+    // Create the .josh-link.toml content
+    let link_content = toml::to_string(&link_file).context("Failed to serialize link file")?;
+
+    // Create the blob for the .josh-link.toml file
+    let link_blob = repo
+        .blob(link_content.as_bytes())
+        .context("Failed to create blob")?;
+
+    // Create the path for the .josh-link.toml file
+    let link_path = std::path::Path::new(&normalized_path).join(".josh-link.toml");
+
+    // Insert the .josh-link.toml file into the tree
+    let new_tree = tree::insert(&repo, &head_tree, &link_path, link_blob, 0o0100644)
+        .map_err(from_josh_err)
+        .context("Failed to insert link file into tree")?;
+
+    // Create a new commit with the updated tree
+    let signature = if let Ok(time) = std::env::var("JOSH_COMMIT_TIME") {
+        git2::Signature::new(
+            "JOSH",
+            "josh@josh-project.dev",
+            &git2::Time::new(time.parse().context("Failed to parse JOSH_COMMIT_TIME")?, 0),
+        )
+        .context("Failed to create signature")?
+    } else {
+        repo.signature().context("Failed to get signature")?
+    };
+
+    let new_commit = repo
+        .commit(
+            None, // Don't update any reference
+            &signature,
+            &signature,
+            &format!("Add link: {}", normalized_path),
+            &new_tree,
+            &[&head_commit],
+        )
+        .context("Failed to create commit")?;
+
+    // Apply the :link filter to the new commit
+    let link_filter = josh_core::filter::parse(":link=snapshot")
+        .map_err(from_josh_err)
+        .context("Failed to parse :link filter")?;
+
+    // Load the cache and create transaction
+    let repo_path = repo.path().parent().unwrap();
+
+    josh_core::cache_sled::sled_load(&repo_path).unwrap();
```

Comment:
In general I don't like unwrap(). Maybe it's okay in a CLI tool, but why not handle the error like for the other error cases, adding some context to what happened?